dotnet/src/Connectors/Connectors.Memory.Milvus/MilvusMemoryStore.cs

```diff
@@ -0,0 +1,429 @@
+    {
+        // TODO: Milvus v2.3.0 will have a 1st-class upsert API which we should use.
+        // In the meantime, we do delete+insert, following the Python connector's example.
+
+        StringBuilder idString = new();
+
+        List<bool> isReferenceData = new();
+        List<string> externalSourceNameData = new();
+        List<string> idData = new();
+        List<string> descriptionData = new();
+        List<string> textData = new();
+        List<string> additionalMetadataData = new();
+        List<ReadOnlyMemory<float>> embeddingData = new();
+        List<string> keyData = new();
+        // List<DateTime> timestampData = new();
+
+        foreach (MemoryRecord record in records)
+        {
+            var metadata = record.Metadata;
+
+            if (idString.Length > 0)
+            {
+                idString.Append(',');
+            }
+
+            idString.Append('"').Append(metadata.Id).Append('"');
+
+            isReferenceData.Add(metadata.IsReference);
+            externalSourceNameData.Add(metadata.ExternalSourceName);
+            idData.Add(record.Metadata.Id);
+            descriptionData.Add(metadata.Description);
+            textData.Add(metadata.Text);
+            additionalMetadataData.Add(metadata.AdditionalMetadata);
+            embeddingData.Add(record.Embedding);
+            keyData.Add(record.Key);
+            // timestampData.Add(record.Timestamp ?? default);
+        }
+
+        MilvusCollection collection = this._milvusClient.GetCollection(collectionName);
+        await collection.DeleteAsync($"{IdFieldName} in [{idString}]", cancellationToken: cancellationToken).ConfigureAwait(false);
+
+        FieldData[] fieldData =
+        {
+            FieldData.Create(IsReferenceFieldName, isReferenceData),
+            FieldData.Create(ExternalSourceNameFieldName, externalSourceNameData),
+            FieldData.Create(IdFieldName, idData),
+            FieldData.Create(DescriptionFieldName, descriptionData),
+            FieldData.Create(TextFieldName, textData),
+            FieldData.Create(AdditionalMetadataFieldName, additionalMetadataData),
+            FieldData.CreateFloatVector(EmbeddingFieldName, embeddingData),
+            FieldData.Create(KeyFieldName, keyData)
+            // TODO: Timestamp
+        };
+
+        MutationResult result = await collection.InsertAsync(fieldData, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+        Debug.Assert(result.Ids.StringIds is not null);
+        foreach (var id in result.Ids.StringIds)
+        {
+            yield return id;
+        }
+    }
+
+    public async Task<MemoryRecord?> GetAsync(string collectionName, string key, bool withEmbedding = false, CancellationToken cancellationToken = default)
+    {
+        IAsyncEnumerator<MemoryRecord> enumerator = this.GetBatchAsync(collectionName, new[] { key }, withEmbedding, cancellationToken).GetAsyncEnumerator(cancellationToken);
```

Comment:
We should dispose of this when done. Rather than direct use of GetEnumerator, maybe structure it as an `await foreach`?
```C#
await foreach (MemoryRecord record in this.GetBatchAsync(collectionName, new[] { key }, withEmbedding, cancellationToken)
{
    return record;
}

return null;
````